src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -36,73 +37,106 @@ class SaaSConnector(BaseConnector[AuthenticatedClient]):
 
     def __init__(self, configuration: ConnectionConfig):
         super().__init__(configuration)
-        self.secrets = configuration.secrets
-        self.saas_config = configuration.get_saas_config()
-        self.client_config = self.saas_config.client_config  # type: ignore
-        self.endpoints = self.saas_config.top_level_endpoint_dict  # type: ignore
-        self.collection_name: Optional[str] = None
-        self.privacy_request: Optional[PrivacyRequest] = None
+        required_saas_config = configuration.get_saas_config()
+        assert required_saas_config is not None
+        self.saas_config = required_saas_config
+        self.endpoints = self.saas_config.top_level_endpoint_dict
+        self.secrets = cast(Dict, configuration.secrets)
+        self.current_collection_name: Optional[str] = None
+        self.current_privacy_request: Optional[PrivacyRequest] = None
+        self.current_saas_request: Optional[SaaSRequest] = None
 
     def query_config(self, node: TraversalNode) -> SaaSQueryConfig:
         """
         Returns the query config for a given node which includes the endpoints
         and connector param values for the current collection.
         """
-        # store collection_name for logging purposes
-        self.collection_name = node.address.collection
+        privacy_request = self.current_privacy_request
+        assert privacy_request is not None
         return SaaSQueryConfig(
             node,
             self.endpoints,
-            self.secrets,  # type: ignore
-            self.saas_config.data_protection_request,  # type: ignore
-            self.privacy_request,  # type: ignore
+            self.secrets,
+            self.saas_config.data_protection_request,
+            privacy_request,
         )
 
+    def get_client_config(self) -> ClientConfig:
+        """Utility method for getting client config according to the current class state"""
+        saas_config_client_config = self.saas_config.client_config
+        current_request_client_config = (
+            self.current_saas_request.client_config
+            if self.current_saas_request and self.current_saas_request.client_config
+            else None
+        )
+        return current_request_client_config or saas_config_client_config
+
+    def get_rate_limit_config(self) -> Optional[RateLimitConfig]:
+        """Utility method for getting rate limit config according to the current class state"""
+        saas_config_rate_limit_config = self.saas_config.rate_limit_config
```

Comment:
this is nice - concise, readable, and straightforward logic.

would it be worth adding in a small unit test that just verifies the behavior here with different configurations? it may be overkill since the logic isn't very complex, but if this gets more complex or something changes moving forward it'd be good to have a test in place. it'd also just demonstrate the config functionality in a bit more depth than the docs (which are great!) can do.

Reply:
That's a good idea. I added some tests. We already had some in place for client config and these are pretty similar.